src/bindgen/ir/enumeration.rs

```diff
@@ -642,12 +650,14 @@ impl Source for Enum {
                 }
 
                 out.open_brace();
-            }
 
-            // Emit the body_prepend section, if relevant
-            if let Some(body) = config.export.pre_body(&self.path) {
-                out.write_raw_block(body);
-                out.new_line();
+                // Emit the pre_body section, if relevant
```

Comment:
We should probably remove this C-specific path. Adding stuff to C is not so useful, and this is adding the body to the struct that has the tag instead of the enum, right?

Reply:
```
typedef struct MyFancyEnum_Prepended {
  #ifdef __cplusplus
    inline void wohoo();
  #endif
  MyFancyEnum_Prepended_Tag tag;
  union {
    Bar_Prepended_Body bar_prepended;
    Baz_Prepended_Body baz_prepended;
  };
} MyFancyEnum_Prepended;
```

This is how it looks for every C output. This is the correct place, right? Post_body puts it in the same place.

IMO we should try to keep consistency between the pre_body and post_body segments wherever possible, and the C tagged structs get post_body, so it would make sense forthem to get pre_body too.

Reply:
Yeah, good point, I was mistaken here. Forgot about the anonymous union thing.